martus-meta/source/org/martus/meta/TestSha1ToSha2Change.java

```diff
@@ -132,6 +149,94 @@ public void testSha1ClientUploadBulletinToSha2Server() throws Exception
 		TRACE_END();
 	}
 
+	public void testSha1ClientShareBulletinWithSha1Contact() throws Exception
+	{
+		TRACE_BEGIN("testSha1ClientShareBulletinWithSha1Contact");
+
+		addHqForClient(appSha1, hqAppSha1.getAccountId());
+
+		Bulletin b1 = createTestBulletin(appSha1);
+		Bulletin b2 = createTestBulletin(appSha1);
+
+		assertEquals("failed upload1?", NetworkInterfaceConstants.OK, sha1AppUploader.uploadBulletin(b1));
+		assertEquals("failed upload2?", NetworkInterfaceConstants.OK, sha1AppUploader.uploadBulletin(b2));
+
+		Vector uidList = new Vector();
+		uidList.add(b1.getUniversalId());
+		uidList.add(b2.getUniversalId());
+		Retriever retriever = new Retriever(hqAppSha1, null);
+		retriever.retrieveBulletins(uidList, hqAppSha1.createFolderRetrievedFieldOffice());
+		assertEquals("retrieve field office bulletins failed?", NetworkInterfaceConstants.OK, retriever.getResult());
+
```

Comment:
The setting up of uidList and assertEqualscan be extracted out into a verify method.